src/providers/delimitedtext/qgsdelimitedtextsourceselect.cpp

```diff
@@ -46,6 +46,10 @@ QgsDelimitedTextSourceSelect::QgsDelimitedTextSourceSelect( QWidget *parent, Qt:
   setupButtons( buttonBox );
   connect( buttonBox, &QDialogButtonBox::helpRequested, this, &QgsDelimitedTextSourceSelect::showHelp );
 
+  // force sample table to show at least 10 rows, even if we have to vertically scroll the widget. Allowing it to shrink
+  // any smaller to avoid a vertical scroll bar makes the sample preview unusable.
+  tblSample->setMinimumHeight( QFontMetrics( font() ).height() * 10 );
```

Comment:
Is the 10 rows minimum note correct because I can get it shrunk down to 5/6
![image](https://user-images.githubusercontent.com/7983394/182098994-cf3caa73-0f70-4ab4-8de0-6ae7e940e1a8.png)
but yes, it expands now and it is way better. Thanks
![image](https://user-images.githubusercontent.com/7983394/182099719-15c96ade-7f03-48ad-a635-ac910dd2ae6f.png)